packages/components/tag/src/tag.js

```diff
@@ -0,0 +1,93 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper } from '@tradeshift/elements';
+import '@tradeshift/elements.icon';
+import css from './tag.css';
+
+export class TSTag extends TSElement {
+	static get styles() {
+		return [TSElement.styles, unsafeCSS(css)];
+	}
+
+	static get properties() {
+		return {
+			/** Text direction: 'rtl' or 'ltr' */
+			dir: { type: String, reflect: true },
+			/** Type: success, warning, warning-lite, danger, blue, blue-lite */
+			type: { type: String, reflect: true },
+			/** Can the item be deleted? */
+			deletable: { type: Boolean, reflect: true },
+			/** Should the tag look like clicking it triggers an action? */
+			clickable: { type: Boolean, reflect: true },
+			/** Is the item locked? */
+			locked: { type: Boolean, reflect: true },
+			/** Array of labels or 'keys' */
+			labels: { type: Array, reflect: true },
+			/** Array of values */
+			values: { type: Array, reflect: true },
+			/** Show busy/loading animation */
+			busy: { type: Boolean, reflect: true }
+		};
+	}
+
+	get direction() {
+		return this.dir || this.bodyDir;
+	}
+
+	handleDeleteEvent(event) {
+		/** constraint event; do not trigger events up the DOM tree */
+		event.stopPropagation();
+		if (!this.busy) {
+			this.dispatchCustomEvent('delete-tag', event);
+		}
+	}
+
+	handleClickEvent(event) {
+		if (!this.busy) {
+			this.dispatchCustomEvent('click-tag', event);
+		}
+	}
+
+	_hasBothLabelsAndValues() {
+		return Array.isArray(this.labels) && this.labels.length && Array.isArray(this.values) && this.values.length;
+	}
+
+	get classes() {
+		const classList = ['container', this._hasBothLabelsAndValues() && 'label-value'];
+		return classList.filter(classname => classname).join(' ');
+	}
+
+	get icons() {
+		if (this.deletable && !this.locked) {
+			return html`
+				<span class="icon action" @click=${this.handleDeleteEvent}>
+					<ts-icon icon="close" size="large"></ts-icon>
+				</span>
+			`;
+		}
+
+		if (this.locked) {
+			return html`
+				<span class="icon">
+					<ts-icon icon="lock" size="large"></ts-icon>
+				</span>
+			`;
+		}
+
```

Comment:
by swapping their places, we can get rid of the `!this.locked` condition on the first `if`